pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1741,6 +1741,14 @@ type EndpointAddress struct {
 
 // ProtocolSpec represents a Ceph Object Store protocol specification
 type ProtocolSpec struct {
+	// Represents RGW 'rgw_enable_apis' config option. See: https://docs.ceph.com/en/reef/radosgw/config-ref/#confval-rgw_enable_apis
+	// If no value provided then all APIs will be enabled: s3, s3website, swift, swift_auth, admin, sts, iam, notifications
+	// If enabled APIs are set, all remaining APIs will be disabled.
+	// This option overrides S3.Enabled value.
+	// +optional
+	// +nullable
+	EnableAPIs []ObjectStoreAPI `json:"enableAPIs,omitempty"`
```

Comment:
Does it make sense instead to use something like this:
```go
type ObjectStoreSpec struct{
   Config map[string]string
  ...
```
to be able to set any rgw options?

Reply:
I have had some thoughts to enable something like this. I think it would reasonably be its own separate PR, but I am interested in exploring this. LMK what you think

Reply:
good idea! in this case it should definetely go in a separate PR.